src/Cli/dotnet/commands/dotnet-workload/clean/WorkloadCleanCommand.cs

```diff
@@ -0,0 +1,90 @@
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.Workloads.Workload.Install;
+using Microsoft.DotNet.Workloads.Workload.List;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+#nullable enable
+
+namespace Microsoft.DotNet.Workloads.Workload.Clean
+{
+    internal class WorkloadCleanCommand : WorkloadCommandBase
+    {
+        private readonly bool _cleanAll;
+
+        private readonly ReleaseVersion _sdkVersion;
+        private readonly IInstaller _workloadInstaller;
+        private readonly IWorkloadResolver _workloadResolver;
+
+        public WorkloadCleanCommand(
+            ParseResult parseResult,
+            IReporter? reporter = null,
+            IWorkloadResolver? workloadResolver = null,
+            string? dotnetDir = null,
+            string? version = null,
+            string? userProfileDir = null
+            ) : base(parseResult, reporter: reporter)
+        {
+            _cleanAll = parseResult.GetValue(WorkloadCleanCommandParser.CleanAllOption);
+
+            string? dotnetPath = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
+            if (dotnetPath == null)
+            {
+                throw new GracefulException(String.Format(LocalizableStrings.InvalidWorkloadProcessPath, Environment.ProcessPath ?? "null"));
+            }
+
+            userProfileDir = userProfileDir ?? CliFolderPathCalculator.DotnetUserProfileFolderPath;
+
+            _sdkVersion = WorkloadOptionsExtensions.GetValidatedSdkVersion(parseResult.GetValue(WorkloadUninstallCommandParser.VersionOption), version, dotnetPath, userProfileDir, true);
+            var sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
+
+            var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(dotnetPath, _sdkVersion.ToString(), userProfileDir);
+            _workloadResolver = workloadResolver ?? WorkloadResolver.Create(workloadManifestProvider, dotnetPath, _sdkVersion.ToString(), userProfileDir);
+            _workloadInstaller = WorkloadInstallerFactory.GetWorkloadInstaller(Reporter, sdkFeatureBand, _workloadResolver, Verbosity, userProfileDir, VerifySignatures, PackageDownloader, dotnetPath);
+        }
+
+        public override int Execute()
+        {
+            ExecuteGarbageCollection();
+            RestoreManifestStates();
+            return 0;
+        }
+
+        private void ExecuteGarbageCollection()
+        {
+            if (_cleanAll)
+            {
+                _workloadInstaller.GarbageCollectInstalledWorkloadPacks(cleanAllPacks: true);
+
+                if (OperatingSystem.IsWindows())
+                {
+                    InstalledWorkloadsCollection vsWorkloads = new();
+                    VisualStudioWorkloads.GetInstalledWorkloads(_workloadResolver, new SdkFeatureBand(_sdkVersion), vsWorkloads);
+
+                    foreach (var vsWorkload in vsWorkloads.AsEnumerable())
```

Comment:
We should decide if we want to clean all feature bands, and then if not, do we still list VS workloads from all feature bands, or just the current? My vote is that we list all of them.